coffee/session.coffee

```diff
@@ -40,20 +40,26 @@ class palava.Session extends @EventEmitter
   #
   reconnect: =>
     @emit 'session_reconnect'
-    @clearConnection()
+    @tearDown()
     @createChannel()
     @createRoom()
     @room.join()
 
   # Reset channel and room
   #
-  clearConnection: =>
+  # @param o [Object] Also release user media
+  #
+  tearDown: (resetUserMedia = false) =>
+    @room?.removeAllListeners()
+    @channel?.removeAllListeners()
     if @channel?.isConnected()
       @room?.leave()
-      @channel.close()
+    @channel.close()
```

Comment:
if you write @channel?.isConnected() above, this should be @channel?.close(), right?